x-pack/plugins/apm/server/lib/services/get_service_transaction_groups/index.ts

```diff
@@ -0,0 +1,322 @@
+                    ),
+                  },
+                },
+                [EVENT_OUTCOME]: {
+                  terms: {
+                    field: EVENT_OUTCOME,
+                  },
+                  aggs: {
+                    transaction_count: {
+                      value_count: {
+                        field: getTransactionDurationFieldForAggregatedTransactions(
+                          searchAggregatedTransactions
+                        ),
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const deltaAsMinutes = (end - start) / 1000 / 60;
+
+  const transactionGroupsWithAllData = sortedAndSlicedTransactionGroups.map(
+    (transactionGroup) => {
+      const groupBucket = timeseriesResponse.aggregations?.transaction_groups.buckets.find(
+        ({ key }) => key === transactionGroup.name
+      );
+
+      const transactionTypes =
+        groupBucket?.transaction_types.buckets.map(
+          (bucket) => bucket.key as string
+        ) ?? [];
+
+      const transactionType =
+        transactionTypes.find(
+          (type) =>
+            type === TRANSACTION_PAGE_LOAD || type === TRANSACTION_REQUEST
+        ) ?? transactionTypes[0];
+
+      const timeseriesBuckets = groupBucket?.timeseries.buckets ?? [];
+
+      return timeseriesBuckets.reduce(
+        (prev, point) => {
+          return {
+            ...prev,
+            latency: {
+              ...prev.latency,
+              timeseries: prev.latency.timeseries.concat({
+                x: point.key,
+                y: point.avg_latency.value,
+              }),
+            },
+            traffic: {
+              ...prev.traffic,
+              timeseries: prev.traffic.timeseries.concat({
+                x: point.key,
+                y: point.transaction_count.value / deltaAsMinutes,
+              }),
+            },
+            error_rate: {
+              ...prev.error_rate,
+              timeseries: prev.error_rate.timeseries.concat({
+                x: point.key,
+                y:
+                  point.transaction_count.value > 0
+                    ? (point[EVENT_OUTCOME].buckets.find(
+                        (outcomeBucket) =>
+                          outcomeBucket.key === EventOutcome.failure
+                      )?.transaction_count.value ?? 0) /
+                      point.transaction_count.value
+                    : null,
+              }),
+            },
+          };
+        },
+        {
+          name: transactionGroup.name,
+          transaction_type: transactionType,
+          latency: {
+            value: transactionGroup.latency,
+            timeseries: [] as Array<{ x: number; y: number | null }>,
+          },
+          traffic: {
+            value: transactionGroup.traffic,
+            timeseries: [] as Array<{ x: number; y: number }>,
```

Comment:
Don't we have existing types for timeseries? If not, shouldn't we?

Reply:
We do, but it's pretty heavy. I think I prefer a simple literal type over a shared one that try to conveys much more than just a simple array of x/y values.

Reply:
sgtm 👍